Cython/Compiler/Options.py

```diff
@@ -371,6 +373,8 @@ def normalise_encoding_name(option_name, encoding):
     'iterable_coroutine': ('module', 'function'),
     'trashcan' : ('cclass',),
     'total_ordering': ('cclass', ),
+    'cpp_locals': ('module', 'function', 'cclass'),  # I don't think they make sense in a with_statement
+    'cpp_locals_nocheck': ('module', 'function', 'cclass'),
```

Comment:
I wondering if this should have great granularity (which would mean changing how it's implemented, so it wouldn't be a property of the type) so that the user could control whether individual accesses are checked